src/backend/columnar/columnar_customscan.c

```diff
@@ -773,15 +916,21 @@ CheckPushdownClause(PlannerInfo *root, RelOptInfo *rel, Expr *clause)
 	 * Check that statistics on the Var support the utility of this
 	 * clause.
 	 */
-	if (!CheckVarStats(root, varSide, sortop))
+	float4 absVarCorrelation = 0;
+	if (!CheckVarStats(root, varSide, sortop, &absVarCorrelation))
 	{
 		ereport(ColumnarPlannerDebugLevel,
-				(errmsg("columnar planner: cannot push down clause: "
-						"var attribute %d is uncorrelated", varSide->varattno)));
-		return false;
+				(errmsg(COLUMNAR_CANNOT_PUSHDOWN_EXPR_HEADER_FMT
+						"absolute correlation (%.3f) of var \"%s\" is smaller than "
+						"\"columnar.qual_pushdown_correlation_threshold\" (%.3f)",
```

Comment:
```suggestion
						"absolute correlation (%.3f) of var \"%s\" is smaller than the value configured in "
						"\"columnar.qual_pushdown_correlation_threshold\" (%.3f)",
```

^ this makes the message slightly easier to understand without directly knowing the names of the GUC's